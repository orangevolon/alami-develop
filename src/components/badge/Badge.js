import { withPubSub, withDOM } from "../../hocs";
import "./Badge.scss";

const Badge = ({ src, document, pubSub }) => {
  const element = document.createElement("img");
  element.src = src;
  element.className = "badge__img";

  return element;
};

export default withPubSub(withDOM(Badge));
