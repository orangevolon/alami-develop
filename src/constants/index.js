export * from "./pubSub";
export * from "./scrollWatch";
