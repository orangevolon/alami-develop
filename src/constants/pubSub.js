export const PUBSUB_SCROLL = "PUBSUB_SCROLL";
