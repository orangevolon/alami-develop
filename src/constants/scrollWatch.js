export const SW_TYPE_ENTER = "enter";
export const SW_TYPE_EXIT = "exit";
export const SW_DIR_DOWN = "down";
export const SW_DIR_UP = "up";
