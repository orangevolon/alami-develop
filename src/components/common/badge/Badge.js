import { withDOM, withPubSub } from "@hocs";
import { PUBSUB_SCROLL } from "@constants";
import "./Badge.scss";
import transform from "../../../lib/transform";

const Badge = ({ src, document, window, pubSub }) => {
  const element = document.createElement("img");
  element.src = src;
  element.className = "badge__img";
  element.style.transformOrigin = "center center";

  pubSub.subscribe(PUBSUB_SCROLL, e => {
    element.style.transform = transform(-element.getBoundingClientRect().y)
      .offset(window.innerHeight / 2)
      .offset(-element.getBoundingClientRect().height / 2)
      .scale(5 / window.innerHeight)
      .inverseSquare(1)
      .scale(0.4)
      .offset(0.8)
      .format(val => `translate(-50%, -50%) scale(${val})`);
  });

  return element;
};

export default withDOM(withPubSub(Badge));
