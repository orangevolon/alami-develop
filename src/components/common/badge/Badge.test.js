import Badge from "./Badge";
import { PUBSUB_SCROLL } from "../../../constants";

jest.mock("@hocs", () => ({
  withPubSub: Component => Component,
  withDOM: Component => props =>
    Component({ document: global.document, ...props })
}));

describe("Testing Components/Badge", () => {
  const pubSub = {
    subscribe: jest.fn()
  };

  beforeEach(() => {
    jest.clearAllMocks();
  });

  it("Should render the Badge correctly", () => {
    expect(Badge({ src: "test/path", pubSub })).toMatchSnapshot();
  });

  it("Should call the subscribe once with the right channel", () => {
    Badge({ src: "test/path", pubSub });

    expect(pubSub.subscribe).toBeCalledWith(
      PUBSUB_SCROLL,
      expect.any(Function)
    );
  });
});
