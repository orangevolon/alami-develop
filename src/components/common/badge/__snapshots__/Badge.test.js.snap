// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Testing Components/Badge Should render the Badge correctly 1`] = `
<img
  class="badge__img"
  src="test/path"
  style="transform-origin: center center;"
/>
`;
